drivers/hmis/app/models/hmis/group_viewable_entity.rb

```diff
@@ -0,0 +1,54 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Hmis
+  class GroupViewableEntity < GrdaWarehouse::GroupViewableEntity
+    acts_as_paranoid
+
+    belongs_to :access_group, class_name: '::Hmis::AccessGroup', inverse_of: :group_viewable_entities
+    belongs_to :entity, polymorphic: true
+
+    scope :projects, -> { where(entity_type: Hmis::Hud::Project.sti_name) }
+    scope :organizations, -> { where(entity_type: Hmis::Hud::Organization.sti_name) }
+    scope :data_sources, -> { where(entity_type: GrdaWarehouse::DataSource.sti_name) }
+    scope :project_access_groups, -> { where(entity_type: GrdaWarehouse::ProjectAccessGroup.sti_name) }
+
+    scope :includes_project, ->(project) do
+      where(entity_type: project.class.name, entity_id: project.id).
+        or(includes_data_source(project.data_source)).
+        or(includes_organization(project.organization)).
+        or(where(entity_type: GrdaWarehouse::ProjectAccessGroup.name, entity_id: project.project_groups.pluck(:id)))
+    end
```

Comment:
@eanders

Reply:
Maybe this should be adjusted to match the `includes_data_source` pattern, so `included_in_project_groups(project.project_groups)` or similar?

Reply:
fixed